experimental/uefi/channel/src/lib.rs

```diff
@@ -19,18 +19,53 @@
 
 extern crate alloc;
 
-use alloc::{vec, vec::Vec};
+use alloc::{string::String, vec, vec::Vec};
 use anyhow::Context;
 use ciborium_io::{Read, Write};
 
+pub mod schema {
+    #![allow(clippy::derivable_impls, clippy::needless_borrow)]
+    #![allow(dead_code, unused_imports)]
+
+    include!(concat!(env!("OUT_DIR"), "/schema_generated.rs"));
+    include!(concat!(env!("OUT_DIR"), "/schema_services.rs"));
+}
+
 /// Length of the entire frame, including the header.
 pub type FrameLength = u32;
-const FRAME_BYTES_SIZE: usize = 4;
+const FRAME_LENGTH_SIZE: usize = 4;
+static_assertions::assert_eq_size!([u8; FRAME_LENGTH_SIZE], FrameLength);
+
+/// In request frames this field contains the id of the method that is invoked.
+/// In response frames this field contains the status code of the method invocation.
+pub type Method = u32;
+const METHOD_SIZE: usize = 4;
+static_assertions::assert_eq_size!([u8; METHOD_SIZE], Method);
+
+pub const FRAME_HEADER_SIZE: usize = 8;
+static_assertions::assert_eq_size!(
+    [u8; FRAME_HEADER_SIZE],
+    [u8; FRAME_LENGTH_SIZE + METHOD_SIZE]
+);
 
 pub struct Frame {
+    pub method: Method,
     pub body: Vec<u8>,
 }
 
+impl Frame {
+    /// Returns the size of the Frame encoded as bytes. On the wire each frame
+    /// is prefixed with its length. This size includes all of the frame,
+    /// including the length prefix. The length is intentionally not stored as a
+    /// value in the frame struct itself, as this may lead to inconsistent
+    /// information if other values are mutated.
+    fn len(&self) -> anyhow::Result<FrameLength> {
+        FrameLength::try_from(FRAME_HEADER_SIZE + self.body.len())
```

Comment:
Why doesn't `Frame::len` include the `METHOD_SIZE` ?

Reply:
`FRAME_HEADER_SIZE` contains `METHOD_SIZE `, see L45 :)